cpp/ql/lib/semmle/code/cpp/models/implementations/Pure.qll

```diff
@@ -26,22 +27,43 @@ private class PureStrFunction extends AliasFunction, ArrayFunction, TaintFunctio
   }
 
```

Comment:
We could add a helper predicate for the locale argument check?
```suggestion

  /** Holds if `i` is a locale parameter that does not carry taint. */
  private predicate isLocaleParameter(ParameterIndex i) {
    this.getName().matches("%\\_l") and i + 1 = this.getNumberOfParameters()
  }
  
```

Reply:
Good idea! Fixed in 7792839a25348cd909074da2b5fa1c11484d2ad9